_posts/2018-02-12-deep-learning-popularity.md

```diff
@@ -0,0 +1,59 @@
+
+In the case of neural networks, what is inside the device is a large amount of interconnected simple processing units called neurons. A neuron takes a number, squeezes it through some non-linear function and then outputs the result. In a deep neural network, the neurons are organized into layers that succeed each other. The input signal is first sent to the first layer of neurons, which send their outputs to all the neurons in the next layer. This process continues layer after layer until the output layer is reached. The construct is inspired by biological neurons, the main components of the central nervous system.
+
+The connections between neurons are also given so called weights, which are basically little valves that determine how much of each input the unit propagates up the network. Adjusting these weights is what actually happens during the training process and what allows neural networks to fit specific problems. A deep neural network could contain millions of weights, so the good news is that adjusting the weights can be automated efficiently (using backpropagation and gradient descent methods).
+
+### Advantages of neural networks
+
+The idea of training a machine to transform numerical representations of inputs to outputs applies to most machine learning models, so what makes neural networks work so well? Three reasons come to mind.
+
+First, the structure of a neural network is specified only very broadly before the model is trained, which gives a lot of room for the model to adjust during training. In statistical terms, large neural networks can be thought of as being somewhere in between parametric and nonparametric models. In a parametric model, for instance a traditional regression, the number of parameters in the model is strictly determined before fitting the model. In a nonparametric model, the model structure is determined more broadly, and the training process can adjust the number of parameters as well as their values. Thus, in a nonparametric model, there is more freedom for the model structure to adjust to the problem being solved. In the case of neural networks, the number of parameters (weights) is strictly determined beforehand, which would imply they are parametric models. However, the number of weights can be enormous and the training process could allow many of the weights to zero out, effectively blocking certain paths through the network. For these reasons, deep and broad neural networks resemble nonparametric models in practice. The nonparametric nature gives neural networks structural freedom to adapt to many kinds of problems. 
+
+Second, since neural networks comprise of chained little functions that perform nonlinear transformations at each step, they are inherently nonlinear models. This allows them to model many problems better since many real-world relationships are nonlinear. (For example, the area of a square shaped field increases exponentially instead of linearly as its width increases.)
```

Comment:
If we want to get really nit-picky: [~comprise of~](https://www.npr.org/2015/03/12/392568604/dont-you-dare-use-comprised-of-on-wikipedia-one-editor-will-take-it-out) either “comprise” (without “of”) or “consist of”.